File: Framework.D-2015/Framework.D-2015/Seguridad/Hash.vb
Imports System.Security.Cryptography
Imports System.Text

Namespace Seguridad

    Public Class Hash
        Public Shared Function getSHA256(ByVal contraseña As String) As String
            Dim sha256 As SHA256 = SHA256Managed.Create()
            Dim encoding As ASCIIEncoding = New ASCIIEncoding()
            Dim stream As Byte() = Nothing
            Dim sb As StringBuilder = New StringBuilder()
            stream = sha256.ComputeHash(encoding.GetBytes(contraseña))

            For i As Integer = 0 To stream.Length - 1
                sb.AppendFormat("{0:x2}", stream(i))
            Next

            Return sb.ToString()
        End Function
    End Class
End Namespace

File: Framework.D-2015/Framework.D-2015/Seguridad/Hash.cs
using System.Security.Cryptography;
using System.Text;

namespace Framework.D_2015.Seguridad
{
    public class Hash
    {
        public static string getSHA256(string contraseña)
        {
            var sha256 = SHA256.Create();
            var encoding = new ASCIIEncoding();
            byte[] stream = null;
            var sb = new StringBuilder();
            stream = sha256.ComputeHash(encoding.GetBytes(contraseña));
            for (int i = 0, loopTo = stream.Length - 1; i <= loopTo; i++)
                sb.AppendFormat("{0:x2}", stream[i]);
            return sb.ToString();
        }
    }
}